### PR TITLE
bucket-sort

### PR DESCRIPTION
**Pull request by flow6979**

### CODE REVIEW

**Comment by flow6979** [comment]
Hey @wingkwong please review my PR and merge it or suggest some changes if you want

**Review by wingkwong — CHANGES_REQUESTED** [review]
Please see [here](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md#tutorials) first..

**Comment by flow6979** [comment]
HEY @wingkwong I have done according to what you said. still if you want some change please ping me.
if my PR is fine please merge it.
Thank you.

**Comment by flow6979** [comment]
hey brother @wingkwong please review it

File: tutorials/basic-topics/sorting/bucket-sort.md [hunk 1]
```diff
@@ -1,14 +1,95 @@
...
+* Sort the items in each bucket 
+* Merge the buckets - can just concatenate them (gathering phase)
+
+A generalization of counting sort
+
+The values in bucket X must be greater than the values in bucket X - 1 and less than the values in bucket X + 1  
+
+This means that the hash function we use must meet this requirement
+
+```C++
```
Comment (wingkwong) [inline tutorials/basic-topics/sorting/bucket-sort.md:47]: 1. wrong format. Use `<Tabs>` to wrap it.
2. use a LC problem to walk through the topic.

File: tutorials/basic-topics/sorting/bucket-sort.md [hunk 2]
```diff
@@ -1,14 +1,95 @@
...
+}
+
+```
+
+export const suggestedProblems = [
+  {
+    "problemName": "0347 - Top K Frequent Elements",
+    "difficulty": "Medium",
+    "leetCodeLink": "https://leetcode.com/problems/top-k-frequent-elements/",
+    "solutionLink": "../../../solutions/0300-0399/0347 - Top K Frequent Elements"
```
Comment (wingkwong) [inline tutorials/basic-topics/sorting/bucket-sort.md:87]: i don't think this link would work. Have you ever verified it once before requesting review?

File: tutorials/basic-topics/sorting/bucket-sort.md [hunk 3]
```diff
@@ -1,14 +1,95 @@
 ---
-title: 'Bucket Sort'
-description: ''
+title: 'The Bucket Sort Algorithm'
+description: 'bucket sort is one of the famous sorting algorithm.'
```
Comment (wingkwong) [inline tutorials/basic-topics/sorting/bucket-sort.md:3]: that is a bad description

File: tutorials/basic-topics/sorting/bucket-sort.md [hunk 4]
```diff
@@ -1,14 +1,95 @@
 ---
-title: 'Bucket Sort'
-description: ''
+title: 'The Bucket Sort Algorithm'
```
Comment (wingkwong) [inline tutorials/basic-topics/sorting/bucket-sort.md:2]: revert to Bucket Sort please

File: solutions/0300-0399/0347 - Top K Frequent Elements.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+* 1 <= nums.length <= 105
+* -104 <= nums[i] <= 104
+* k is in the range [1, the number of unique elements in the array].
+* It is guaranteed that the answer is unique.
+
+## Approach 1:  bucket sort
+
+<SolutionAuthor name="@flow6979"/>
+
+```cpp
```
Comment (wingkwong) [inline solutions/0300-0399/0347 - Top K Frequent Elements.md:40]: wrong format. Use <Tabs>. See `CONTRIBUTING.md` first please.

File: tutorials/basic-topics/sorting/bucket-sort.md [hunk 6]
```diff
@@ -1,14 +1,95 @@
...
+
+* Distribute the items into buckets based on their hashed values (scattering phase)
+* Sort the items in each bucket 
+* Merge the buckets - can just concatenate them (gathering phase)
+
+A generalization of counting sort
+
+The values in bucket X must be greater than the values in bucket X - 1 and less than the values in bucket X + 1  
+
+This means that the hash function we use must meet this requirement
```
Comment (wingkwong) [inline tutorials/basic-topics/sorting/bucket-sort.md:45]: What's the point of copying the exact content from https://github.com/akarazhev/algorithms-structures-patterns/blob/master/algorithms/doc/bucket-sort.md ?

**Review by wingkwong — CHANGES_REQUESTED** [review]
DON'T COPY CONTENT FROM OTHER SITES.

**Comment by flow6979** [comment]
> DON'T COPY CONTENT FROM OTHER SITES.

brother i have not copied code and explanation.. but things like time complexity, you cant do anything with that, it needs to be seen from somewhere.

IF YOU WANT SOMETHING SPECIFIC TO BE CHANGED..tell me

**Comment by wingkwong** [comment]
First, that's from other people work. You didn't even give a credit. Time Complexity and Space Complexity are fine. You're just copying the exact content even with the author's bad formatting. 

Please redo it. I won't accept this kind of work.

**Comment by flow6979** [comment]
ok i will do it myself again

**Comment by wingkwong** [comment]
Any update?

**Comment by wingkwong** [comment]
any update?

**Comment by wingkwong** [comment]
Closing due to inactivity.

**Pull request closed without merging**